### PR TITLE
Bugfix/pivotal 172082802 file list output

### PR DESCRIPTION
**Pull request by Juan-EBI**
- Added copy and move support

### CODE REVIEW

**Comment by Juan-EBI** [comment]
@jhoanmanuelms note that there is still pending pt submit  update.

File: commons/commons-util/src/test/kotlin/ebi/ac/uk/io/NfsFileUtilsTest.kt [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        temporaryFolder.clean()
+    }
+
+    @Nested
+    inner class MoveFile {
+
+        @Nested
+        inner class WhenFile {
+            @Test
+            fun whenExistTarget() {
```
Comment (jhoanmanuelms) [inline commons/commons-util/src/test/kotlin/ebi/ac/uk/io/NfsFileUtilsTest.kt:26]: typo: whenTargetExists

File: commons/commons-util/src/test/kotlin/ebi/ac/uk/io/NfsFileUtilsTest.kt [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                val another = temporaryFolder.createFile("two.txt", "two")
+
+                NfsFileUtils.moveFile(file, another)
+
+                assertThat(temporaryFolder.root.resolve("one.txt")).doesNotExist()
+                assertThat(temporaryFolder.root.resolve("two.txt")).hasContent("one")
+            }
+
+            @Test
+            fun whenNotExistTarget() {
```
Comment (jhoanmanuelms) [inline commons/commons-util/src/test/kotlin/ebi/ac/uk/io/NfsFileUtilsTest.kt:37]: typo: whenNoTargetExists

File: commons/commons-util/src/test/kotlin/ebi/ac/uk/io/NfsFileUtilsTest.kt [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+                assertThat(temporaryFolder.root.resolve("new.txt")).hasContent("one")
+            }
+        }
+
+        @Nested
+        inner class WhenFolder {
+
+            @Test
+            fun whenTargetFolderExist() {
```
Comment (jhoanmanuelms) [inline commons/commons-util/src/test/kotlin/ebi/ac/uk/io/NfsFileUtilsTest.kt:51]: typo: exitsts

File: commons/commons-util/src/test/kotlin/ebi/ac/uk/io/NfsFileUtilsTest.kt [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                targetDirectory.addNewFile("one.txt")
+
+                NfsFileUtils.moveFile(tempDir, targetDirectory)
+
+                assertThat(targetDirectory).isDirectory()
+                assertThat(targetDirectory.list()).containsExactly("two.txt")
+            }
+
+            @Test
+            fun whenNoTargetFolder() {
```
Comment (jhoanmanuelms) [inline commons/commons-util/src/test/kotlin/ebi/ac/uk/io/NfsFileUtilsTest.kt:65]: typo: whenNoTargetFolderExists

File: commons/commons-util/src/test/kotlin/ebi/ac/uk/io/NfsFileUtilsTest.kt [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+                assertThat(target).isDirectory()
+                assertThat(target.list()).containsExactly("two.txt")
+            }
+        }
+    }
+
+    @Nested
+    inner class DeleteFolder {
+        @Test
+        fun deleteFolderWhenFileExist() {
```
Comment (jhoanmanuelms) [inline commons/commons-util/src/test/kotlin/ebi/ac/uk/io/NfsFileUtilsTest.kt:81]: typo: exists

File: commons/commons-util/src/test/kotlin/ebi/ac/uk/io/NfsFileUtilsTest.kt [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+            val file = temporaryFolder.createFile("one.txt", "content.exist")
+
+            NfsFileUtils.deleteFolder(file)
+            assertThat(file).doesNotExist()
+        }
+
+        @Test
+        fun deleteFolderWhenNotEmptyFolder() {
```
Comment (jhoanmanuelms) [inline commons/commons-util/src/test/kotlin/ebi/ac/uk/io/NfsFileUtilsTest.kt:89]: I'd suggest: deleteNotEmptyFolder

File: commons/commons-util/src/test/kotlin/ebi/ac/uk/io/NfsFileUtilsTest.kt [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+            val file = tempDir.resolve("two.txt")
+            file.createNewFile()
+
+            NfsFileUtils.deleteFolder(tempDir)
+            assertThat(tempDir).doesNotExist()
+        }
+
+        @Test
+        fun deleteFolderWhenNotExist() {
+            val tempDir = temporaryFolder.createDirectory("dir-example")
```
Comment (jhoanmanuelms) [inline commons/commons-util/src/test/kotlin/ebi/ac/uk/io/NfsFileUtilsTest.kt:100]: This is not necessary, we just need to call the method with a folder that doesn't exist

File: submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/integration/PersistenceContextImpl.kt [hunk 8]
```diff
@@ -38,15 +39,17 @@ open class PersistenceContextImpl(
...
+    override fun saveSubmission(saveRequest: SaveRequest): ExtSubmission {
+        val (submission, submitter, mode) = saveRequest
+        return lockExecutor.executeLocking(submission.accNo) { saveSubmission(submission, submitter, mode) }
+    }
 
-            val newSubmission = saveNewVersion(submission, submitter)
-            filePersistenceService.persistSubmissionFiles(submission)
-            newSubmission
-        }
+    private fun saveSubmission(submission: ExtSubmission, submitter: User, mode: FileMode): ExtSubmission {
```
Comment (jhoanmanuelms) [inline submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/integration/PersistenceContextImpl.kt:47]: Not sure why we need this method. I think it looks clear just executing this in the `executeLocking` closure

File: submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/service/FilePersistenceService.kt [hunk 9]
```diff
@@ -2,62 +2,89 @@ package ac.uk.ebi.biostd.persistence.service
...
-        val submissionPath = folderResolver.getSubmissionFolder(submission.relPath)
+        val submissionPath = folderResolver.getSubmissionFolder(relPath)
 
         FileUtils.writeStringToFile(submissionPath.resolve("$outputFileName.json").toFile(), json, Charsets.UTF_8)
         FileUtils.writeStringToFile(submissionPath.resolve("$outputFileName.xml").toFile(), xml, Charsets.UTF_8)
         FileUtils.writeStringToFile(submissionPath.resolve("$outputFileName.pagetab.tsv").toFile(), tsv, Charsets.UTF_8)
     }
 
-    private fun persistFiles(submission: ExtSubmission) {
+    private fun copyFiles(submission: ExtSubmission) {
```
Comment (jhoanmanuelms) [inline submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/service/FilePersistenceService.kt:58]: :+1:

File: submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/service/FilePersistenceService.kt [hunk 10]
```diff
@@ -2,62 +2,89 @@ package ac.uk.ebi.biostd.persistence.service
...
+        processFiles(submission, this::copy)
+    }
+
+    private fun moveFiles(submission: ExtSubmission) {
```
Comment (jhoanmanuelms) [inline submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/service/FilePersistenceService.kt:62]: Couldn't this be one-liners?

File: submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/service/FilePersistenceService.kt [hunk 11]
```diff
@@ -2,62 +2,89 @@ package ac.uk.ebi.biostd.persistence.service
...
+    private fun copy(extFile: ExtFile, file: File) = replaceFile(extFile.file, file.resolve(extFile.fileName))
+    private fun move(file: ExtFile, path: File) = NfsFileUtils.moveFile(file.file, path.resolve(file.fileName))
+
     private fun getSubmissionFolder(relPath: String): File {
         val submissionFolder = folderResolver.getSubmissionFolder(relPath).toFile()
         submissionFolder.mkdirs()
         return submissionFolder
     }
 
     private fun createTempFolder(submissionFolder: File, accNo: String): File {
```
Comment (jhoanmanuelms) [inline submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/service/FilePersistenceService.kt:87]: Another one-liner candidate here :grimacing:

File: submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/web/handlers/SubmissionWebHandler.kt [hunk 12]
```diff
@@ -22,32 +22,38 @@ class SubmissionWebHandler(
...
+        submissionService.deleteSubmission(accNo, user)
+
+    fun getSubmissions(user: SecurityUser, filter: SubmissionFilter): List<SimpleSubmission> =
+        submissionService.getSubmissions(user, filter)
+
+    private fun subWithAttributes(subFile: File, attrs: Map<String, String>, source: FilesSource): Submission {
         val submission = serializationService.deserializeSubmission(subFile, source)
-        attachTo?.let { submission.attachTo = attachTo }
-        return submissionService.submit(SubmissionRequest(submission, user, source, FILE))
+        attrs.forEach { submission[it.key] = it.value }
```
Comment (jhoanmanuelms) [inline submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/web/handlers/SubmissionWebHandler.kt:55]: Much better :+1:

File: submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/web/resources/SubmitResource.kt [hunk 13]
```diff
@@ -50,12 +52,15 @@ class SubmitResource(
     fun submitMultipartJson(
         @BioUser user: SecurityUser,
 
-        @ApiParam(name = "Submission", value = "File containing the submission page tab in JSON format")
```
Comment (jhoanmanuelms) [inline submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/web/resources/SubmitResource.kt:53]: Maybe we could just use the constants here?

**Pull request merged**